packages/components/slider/tests/slider.test.tsx

```diff
@@ -87,16 +98,213 @@ describe("<Slider />", () => {
     expect(sliderThumb).toHaveAttribute("aria-readonly", "true")
   })
 
-  test("Slider can have correct step", () => {
+  test("can have correct step", async () => {
     const step = 10
-    render(<Slider data-testid="slider" step={step} />)
+    const defaultValue = 0
+    const { container } = render(
+      <Slider min={0} max={100} defaultValue={defaultValue} step={step} />,
+    )
+
+    const slider = screen.getByRole("slider")
+    const sliderInput = container.getElementsByTagName("input")[0]
+
+    await act(() => fireEvent.keyDown(slider, { key: "ArrowRight" }))
+
+    expect(Number(sliderInput.value)).toBe(defaultValue + step)
+  })
+
+  test("should throw error when max is less than min", () => {
+    const min = 10
+    const max = 5
+
+    const renderWithInvalidProps = () => render(<Slider min={min} max={max} />)
+
+    const consoleSpy = vi.spyOn(console, "error")
+    consoleSpy.mockImplementation(() => {})
+
+    expect(renderWithInvalidProps).toThrow(
+      "Do not assign a number less than 'min' to 'max'",
+    )
+
+    consoleSpy.mockRestore()
+  })
+
+  test("key down should perform correct actions", async () => {
+    const min = 0
+    const max = 100
+    const tenStep = (max - min) / 10
+    const { container } = render(
+      <Slider min={min} max={max} step={10} defaultValue={0} />,
+    )
+
+    const slider = screen.getByRole("slider")
+    const sliderInput = container.getElementsByTagName("input")[0]
 
-    const slider = screen.getByTestId("slider").getElementsByTagName("input")[0]
+    await act(() => fireEvent.keyDown(slider, { key: "ArrowRight" }))
+    expect(Number(sliderInput.value)).toBe(10)
+    await act(() => fireEvent.keyDown(slider, { key: "ArrowLeft" }))
+    expect(Number(sliderInput.value)).toBe(0)
+
+    await act(() => fireEvent.keyDown(slider, { key: "ArrowUp" }))
+    expect(Number(sliderInput.value)).toBe(10)
+    await act(() => fireEvent.keyDown(slider, { key: "ArrowDown" }))
+    expect(Number(sliderInput.value)).toBe(0)
+
+    await act(() => fireEvent.keyDown(slider, { key: "PageUp" }))
+    expect(Number(sliderInput.value)).toBe(0 + tenStep)
+    await act(() => fireEvent.keyDown(slider, { key: "PageDown" }))
+    expect(Number(sliderInput.value)).toBe(0)
+
+    await act(() => fireEvent.keyDown(slider, { key: "Home" }))
+    expect(Number(sliderInput.value)).toBe(min)
+    await act(() => fireEvent.keyDown(slider, { key: "End" }))
+    expect(Number(sliderInput.value)).toBe(max)
+  })
+
+  test("should set isReadOnly to true when focusThumbOnChange is false", () => {
```

Comment:
How about grouping this test case with the "can be readOnly" test case in the same test suite?